terraform/modules/hetzner-kubernetes/load-balancer.resources.tf

```diff
@@ -0,0 +1,80 @@
+resource "hcloud_load_balancer" "lb" {
+  count = var.with_load_balancer ? 1 : 0
+
+  name     = "${var.cluster_name}-lb"
+  location = var.default_location
+
+  load_balancer_type = (
+    length(var.machines) <= 5 ? "lb11" : (
+    length(var.machines) <= 75 ? "lb21" : (
+    length(var.machines) <= 150 ? "lb31" : null))
+  )
```

Comment:
I see that this is a required thing which could potentially be null, does terraform plan fail in this case? What error would the user get?

Reply:
It fails with

```
Error: Required attribute is not set

  on ../modules/hetzner-kubernetes/load-balancer.resources.tf line 7, in resource "hcloud_load_balancer" "lb":
   7:   load_balancer_type = null
```

This is IMHO sufficiently self-explanatory after looking at the corresponding LOCs. Please yell if you disagree!

Reply:
Hmm it is a bit scary though. Say I bump my cluster from 5 to 6 nodes; that will get the load-balancer destroyed and recreated and cause a period of downtime. This sounds less than ideal. Wouldn't this better be something we should bubble up and let the administrator make a concious decision about?

Reply:
You absolutely have a point here, but before drawing any conclusion, I think we should test whether it actually causes destruction and downtime. Additionally, that the LB might get destroyed should be completely transparent in the TF plan. Wouldn't this suffice as a heads-up and pull back if necessary?

Reply:
Yes I suppose it does.

Reply:
I'm convinced :) Consider this resolved